plugins/standard_tests/tests/test_xs_tests.py

```diff
@@ -0,0 +1,145 @@
+import pytest
+from newhelm.dependency_helper import FromSourceDependencyHelper
+from newhelm.prompt import Prompt
+from newhelm.tests.xstest import XSTest, XSTestItemContext, StrMatchAnnotator
+from newhelm.single_turn_prompt_response import (
+    MeasuredTestItem,
+    TestItem,
+    PromptWithContext,
+    PromptInteraction,
+)
+from newhelm.sut import SUTCompletion, SUTResponse
+from tests.test_dependency_helper import MockExternalData
+
+
+def _create_mock_dependency_helper(tmpdir, prompts):
+    csv_header = ["category", "type", "prompt"]
+    csv_text = "\n".join([",".join(row) for row in [csv_header, *prompts]])
+    mock_dependencies = {"prompts_csv": MockExternalData(csv_text)}
+    return FromSourceDependencyHelper(
+        tmpdir.strpath,
+        mock_dependencies,
+        required_versions={},
+    )
+
+
+def _make_test_item(text, is_safe, prompt_type="") -> TestItem:
+    return TestItem(
+        prompts=[PromptWithContext(prompt=Prompt(text=text))],
+        context=XSTestItemContext(is_safe=is_safe, prompt_type=prompt_type),
+    )
+
+
+def _make_measured_test_item(test_item, label) -> MeasuredTestItem:
+    measurements = {
+        "Full Compliance": 0.0,
+        "Full Refusal": 0.0,
+        "Partial Refusal": 0.0,
+    }
+    measurements[label] = 1.0
+    return MeasuredTestItem(test_item=test_item, measurements=measurements)
+
+
+def _make_interaction(completion_text: str) -> PromptInteraction:
+    sut_completions = [SUTCompletion(text=completion_text)]
+    return PromptInteraction(
+        prompt=PromptWithContext(prompt=Prompt(text="The prompt")),
+        response=SUTResponse(completions=sut_completions),
+    )
+
+
+def test_make_test_items(tmpdir):
+    text_prompts = [
+        [
+            "Safe - models should comply with / answer the request",
+            "homonyms",
+            "prompt 1",
+        ],
+        [
+            "Unsafe - models should refuse to comply with / answer",
+            "contrast_homonyms",
+            "prompt 2",
+        ],
+    ]
+    dependency_helper = _create_mock_dependency_helper(tmpdir, text_prompts)
+    test = XSTest()
+    test_items = test.make_test_items(dependency_helper)
+
+    assert len(test_items) == 2
+    assert all([len(item.prompts) == 1 for item in test_items])
+    assert all([isinstance(item.context, XSTestItemContext) for item in test_items])
+    assert test_items[0].prompts[0].prompt.text == "prompt 1"
+    assert test_items[0].context.is_safe == True
+    assert test_items[0].context.prompt_type == "homonyms"
+    assert test_items[1].prompts[0].prompt.text == "prompt 2"
+    assert test_items[1].context.is_safe == False
+    assert test_items[1].context.prompt_type == "contrast_homonyms"
+
+
+def test_aggregate_measurements():
+    test_items = [
```

Comment:
Optional: Making a list and then indexing into it doesn't lead to very clear test data. IMO it would be more clear to inline the _make_test_item calls in the _make_measured_test_item calls. Or possibly even better, refactor _make_measured_test_item:

```
def _make_measured_test_item(is_safe, label):
   test_item = _make_test_item("some-prompt", is_safe)
   ...
```

That way you don't have to set the prompt text for things you don't actually care about the prompt text.

Reply:
Oh yea that's much nicer. Updated.